src/fsharp/TypedTreeOps.fs

```diff
@@ -1819,7 +1842,7 @@ let isStructTy g ty =
     | ValueSome tcref -> 
         isStructTyconRef tcref
     | _ -> 
-        isStructAnonRecdTy g ty || isStructTupleTy g ty
+        isStructAnonRecdTy g ty || isStructTupleTy g ty || isStructErasedUnionTy g ty
```

Comment:
This is a problem - and we shouldn't have the notion of `isStructErasedUnionTy` at all I think

The predicate `isStructTy` is about whether we take the address of something, for example.  But `(int32|int64)` will be represented as a **boxed** type, so we don't need to take the address of anything.

So I think the whole notion of `isStructErasedUnionTy` will not be usable anywhere - the fact that each element is a struct won't change anything in analysis or codegen as far as I'm aware.